File: LinearSearch.java

public class LinearSearch {
    public static int search(int arr[], int key) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == key) {
                return i;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        int[] arr = { 2, 5, 3, 7, 6, 9 };
        int key = 17;
        int result = search(arr, key);
        if (result == -1) {
            System.out.println("the key " + key + " not found!");
        } else {
            System.out.println("The Key " + key + " found at the location " + result);
        }

    }
}

File: JAVA21/begining/DSA in C/WEEK2/linearsearch.c
#include<stdio.h>
int linearserch(int arr[],int target,int n){
    for(int i=0;i<n;i++){
        if(arr[i]==target){
            return i;
        }
    }
    return -1;
}
int main(){
    int n,target;
    printf("enter the size of the array:");
    scanf("%d",&n);
    int arr[n];
    printf("insert the numbers in the array:");
    for(int i=0;i<n;i++){
        scanf("%d",&arr[i]);
    }
    printf("enter the element to search:");
    scanf("%d",&target);
    int result = linearserch(arr,target,n);
    if (result!=-1){
        printf("element %d found at the location %d",target,result+1);
    }else{
        printf("the element %d not found in the list!",target);
    }
    return 0;

}